knowledge_graphs/performance_monitor.py

```
@@ -0,0 +1,497 @@
+                
+                self.system_metrics.append(system_metric)
+                
+                # Record as individual metrics
+                self.record_metric("system_cpu_percent", cpu_percent, "system")
+                self.record_metric("system_memory_percent", memory.percent, "system")
+                self.record_metric("system_memory_available", memory.available, "system")
+                self.record_metric("system_disk_usage_percent", disk.percent, "system")
+                
+                await asyncio.sleep(self.collection_interval)
+                
+            except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)